### PR TITLE
New plugin: Solar-Log

### PR DESCRIPTION
**Pull request by Boernsman**
nymea-plugins pull request checklist:

- [x] Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

- [x] If you added a new plugin, should it be added to nymea-plugins-all or nymea-plugins-maker?

### CODE REVIEW

**Comment by Boernsman** [comment]
Can't test it right now, @simonseres will test it.

**Comment by t-mon** [comment]
removed from experimental for now due to merge conflicts

File: solarlog/integrationpluginsolarlog.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+{
+    QUrl url;
+    url.setHost(thing->paramValue(solarlogThingHostParamTypeId).toString());
+    url.setPath("/getjp");
+    url.setScheme("http");
+    QNetworkRequest request(url);
+    request.setHeader(QNetworkRequest::KnownHeaders::ContentTypeHeader, "application/json");
+    QNetworkReply *reply = hardwareManager()->networkManager()->post(request, QByteArray("{\"801\":{\"170\":null}}"));
+    connect(reply, &QNetworkReply::finished, reply, &QNetworkReply::deleteLater);
+    connect(reply, &QNetworkReply::finished, this, [this, reply, thing]{
```
Comment (mzanetti) [inline solarlog/integrationpluginsolarlog.cpp:93]: connect to thing, not this

File: solarlog/integrationpluginsolarlog.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            }
+            return;
+        }
+
+        thing->setStateValue(solarlogConnectedStateTypeId, true);
+        QByteArray rawData = reply->readAll();
+        QJsonParseError error;
+        QJsonDocument data = QJsonDocument::fromJson(rawData, &error);
+        if (error.error != QJsonParseError::NoError) {
+            qCWarning(dcSolarlog()) << "Received invalide JSON object, try to upgrade the Solarlog firmware. Min Version is 3.5.";
```
Comment (mzanetti) [inline solarlog/integrationpluginsolarlog.cpp:113]: typo, invalid

File: solarlog/integrationpluginsolarlog.json [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                    "createMethods": [ "user", "discovery" ],
+                    "interfaces": [ "connectable" ],
+                    "paramTypes": [
+                        {
+                            "id": "84bd8a41-2411-4bb0-87a9-ab7e01044b10",
+                            "name": "host",
+                            "displayName": "Address",
+                            "type": "QString",
+                            "inputType": "TextLine",
+                            "defaultValue": "http://solar-log"
```
Comment (mzanetti) [inline solarlog/integrationpluginsolarlog.json:24]: hmm... Just asking. the default value is http://solar-log. I assume that solar-log is a local domain name. Setupthing then does `setHost(hostParam)` and `setSchema("http"). Did you test if that doesn't end up being `http://http://solar-log`? I assume QUrl will probably be able to figure this out, but just making sure...

**Pull request merged**